From e8c599d085881b8720773587d4b67a3e598d123b Mon Sep 17 00:00:00 2001
From: Stephane Gasparini <stephane.gasparini@linux.intel.com>
Date: Tue, 9 Feb 2016 17:07:38 +0100
Subject: [PATCH 178/420] kernel/fs: fix I/O wait not accounted for RW O_DSYNC

 When a process is doing Random Write with O_DSYNC flag
 the I/O wait are not accounted in the kernel (get_cpu_iowait_time_us).
 This is preventing the governor or the cpufreq driver to account for
 I/O wait and thus use the right pstate

Signed-off-by: Stephane Gasparini <stephane.gasparini@linux.intel.com>
Signed-off-by: Philippe Longepe <philippe.longepe@linux.intel.com>
Signed-off-by: Jens Axboe <axboe@fb.com>
---
 block/bio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/block/bio.c b/block/bio.c
index 13526f39d1ce..f01b82325194 100644
--- a/block/bio.c
+++ b/block/bio.c
@@ -887,7 +887,7 @@ int submit_bio_wait(int rw, struct bio *bio)
 	bio->bi_private = &ret;
 	bio->bi_end_io = submit_bio_wait_endio;
 	submit_bio(rw, bio);
-	wait_for_completion(&ret.event);
+	wait_for_completion_io(&ret.event);
 
 	return ret.error;
 }
-- 
2.15.0

